model.go

```diff
@@ -26,11 +84,153 @@ func LoadModelFile(path string) (*Model, error) {
 	return LoadModel(fp)
 }
 
-func LoadModel(r io.Reader) (*Model, error) {
+// LoadModel loads a model
+func LoadModel(r io.Reader) (Model, error) {
 	dec := json.NewDecoder(r)
 	var m Model
 	if err := dec.Decode(&m); err != nil {
 		return nil, err
 	}
-	return &m, nil
+	for i := range m {
+		m[i].WeightVec = flattenWeight(m[i].Weight)
+	}
+	return m, nil
+}
+
+func (m Model) newPaddedPixels(p *Pixels) *Pixels {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 11 of function (Model).newPaddedPixels() is high (> 10)